src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -44,40 +44,94 @@ private void buildSimilarityQuery(CriteriaBuilder builder, CriteriaQuery query,
     }
 
     /**
-     * Uses the pg_trgm similarity function as well as any filtering or pagination provided to search for matching Monuments
+     * Creates a PostGIS ST_DWithin query on the Monument's point field and adds it to the specified CriteriaQuery
+     * @param builder The CriteriaBuilder for the query
+     * @param query The CriteriaQuery
+     * @param root The Root associated with the CriteriaQuery
+     * @param latitude The latitude of the point to compare to
+     * @param longitude The longitude of the point to compare to
+     * @param miles - The number of miles from the comparison point to check
+     */
+    private void buildDWithinQuery(CriteriaBuilder builder, CriteriaQuery query, Root root, String latitude,
+                                   String longitude, Integer miles) {
+        String comparisonPointAsString = "POINT(" + longitude + " " + latitude + ")";
+        Integer feet = miles * 5280;
+
+        query.where(
+                builder.equal(
+                        builder.function("ST_DWithin", Boolean.class,
+                                builder.function("ST_Transform", Geometry.class, root.get("point"),
+                                        builder.literal(2877)
+                                ),
+                                builder.function("ST_Transform", Geometry.class,
+                                        builder.function("ST_GeometryFromText", Geometry.class,
+                                                builder.literal(comparisonPointAsString),
+                                                builder.literal(4326)
+                                        ),
+                                        builder.literal(2877)
+                                ),
+                                builder.literal(feet)
+                        ),
+                true)
+        );
+    }
+
```

Comment:
`ST_DWithin` and `ST_Transform` are PostGIS database functions.

Don't even get me started about the `2877` and `4326`...

Reply:
can you make them `public static final int` with a link to one of those sites that explains them